app/models/manageiq/providers/vmware/infra_manager/refresh_parser.rb

```diff
@@ -362,10 +389,36 @@ def self.host_inv_to_switch_hashes(inv)
 
           pnics.each { |pnic| result_uids[:pnic_id][pnic] = new_result unless pnic.blank? }
         end
+
+        unless dvswitch_inv.nil?
+          dvswitch_inv.each do |data|
```

Comment:
You can do `dvswitch_inv.to_miq_a.each` and save the `unless dvswitch_inv.nil?`, can save some indentation.